backend/Origam.DA.Service/AbstractSqlDataService.cs

```diff
@@ -1723,400 +1844,531 @@ private List<ColumnData> GetAllQueryColumns(DataStructureQuery query)
+	                || !(constraint.Fields[0] is FieldMappingItem))
+	                {
+		                continue;
+	                }
+	                var rows = foreignKeys.Tables[0].Select(
+		                "PK_Table = '" 
+		                + (constraint.ForeignEntity as TableMappingItem)
+		                .MappedObjectName 
+		                + "' AND FK_Table = '" 
+		                + table.MappedObjectName 
+		                + "' AND cKeyCol1 = '" 
+		                + (constraint.Fields[0] as FieldMappingItem)
+		                .MappedColumnName 
+		                + "'");
+	                if(columns.Tables[0].Select(
+		                   "TABLE_NAME = '"
+		                   + table.MappedObjectName
+		                   + "' AND COLUMN_NAME = '"
+		                   + (constraint.Fields[0] as FieldMappingItem)
+		                   .MappedColumnName
+		                   + "'").Length <= 0)
+	                {
+		                continue;
+	                }
+	                if(rows.Length == 0)
+	                {
+		                // constraint was not found in the database at all
+		                var result = new SchemaDbCompareResult
+		                {
+			                ResultType = DbCompareResultType
+				                .MissingInDatabase,
+			                ItemName = ConstraintName(
+				                table, constraint),
+			                SchemaItem = table,
+			                SchemaItemType = typeof(DataEntityConstraint),
+			                Script = (DbDataAdapterFactory 
+					                as AbstractSqlCommandGenerator)
+				                .AddForeignKeyConstraintDdl(
+					                table, constraint)
+		                };
+		                results.Add(result);
+	                }
+	                else
+	                {
+		                var constraintEqual = true;
+		                // constraint found in database,
+		                // we check if it has the same fields
+		                foreach(IDataEntityColumn column 
+		                        in constraint.Fields)
+		                {
+			                if(column is FieldMappingItem fieldMappingItem)
+			                {
+				                if(column.ForeignKeyField 
+				                   is FieldMappingItem foreignKeyField)
+				                {
+					                var primaryKey = foreignKeyField
+						                .MappedColumnName;
+					                var foreignKey = fieldMappingItem
+						                .MappedColumnName;
+					                var foundPair = false;
+					                for(var i = 1; i < 17; i++)
+					                {
+						                if((rows[0]["cKeyCol" + i] 
+						                    != DBNull.Value) 
+						                   && ((string)rows[0]["cKeyCol" + i] 
+						                       == foreignKey) 
+						                   && ((string)rows[0]["cRefCol" + i] 
+						                       == primaryKey))
+						                {
```

Comment:
I don't think breaking the statement into 3 more lines made it any more readable. The real problem here is the nesting. Is the result under 80 characters wide anyway?

Reply:
Result was under 80 characters. I've extracted the part of the code to another function to remove nesting.